Grids AI Column Demos: Add Descriptions (#31830)
DevExpress/DevExtreme: apps/demos/Demos/DataGrid/AIColumns/description.md

> DevExtreme DataGrid supports AI columns. These columns augment your data and transform our DataGrid into an AI-powered data exploration tool. This demo implements an AI column (fixed on the right side of the component) to display one additional AI-generated data field. You can also integrate multiple AI columns to expand the scope of generated insights.

Reviewer: I like Sveta's remark about specifying the demo test case 
> After the first paragraph, add a note explaining that the demo includes a default prompt for the AI column. This prompt is used to determine the origin country of a car item. Users can modify this prompt or enter a completely custom one directly through the AI column UI.

<!--split-->

To integrate an AI column, do the following:

- Configure [aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#aiIntegration) (or **columns[]**.**ai**.[aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#aiIntegration)).
- Set a column's [type](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#type) to *"ai"*.
- Specify the column's [name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name).
- Configure **columns[]**.**ai** options, including the [prompt](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#prompt) and generation [mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#mode).

> This demo also configures the [onAIColumnRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIColumnRequestCreating) handler to limit data included in AI requests. The default DataGrid behavior is to include all data from visible rows in AI requests, including data not bound to a column and data of hidden columns. This gives LLMs broader context, but increases the component's use of AI resources. To limit data included in AI requests, modify the **AIColumnRequestCreatingEvent**.[data](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIColumnRequestCreatingEvent/#data) parameter.

Reviewer: AFAIK we have some limitations for AI service that we use in this demo. 
So, it's better to include some Note about it to warning users that the demo is for testing purposes and we've limited tokens...
Please check similar notes in other DX /not-DX products and consult with developer to finalize the text

Reviewer 2: In Blazor, we actually have two notes (see the example "DevExpress AI-powered extensions follow the "bring your own key" principle": https://blazor/serverside-25.2/AI/SemanticSearch#Content ). Please check if the second note is also applicable to your demo.

If yes, you can use them as-is, since they have been already checked a couple of times.

Author: I think we can add the second sentence from that note with a slight modification (specifying "AI REST API"):

"DevExpress does not offer an AI REST API and does not ship any built-in LLMs/SLMs"

The rest of the note, however, is specific to Blazor & AzureOpenAI services and is not applicable.

> DevExtreme DataGrid supports AI columns. These columns augment your data and transform our DataGrid into an AI-powered data exploration tool. This demo implements an AI column (fixed on the right side of the component) to display one additional AI-generated data field. You can also integrate multiple AI columns to expand the scope of generated insights.

Reviewer 2: Let's use some wordings from WN

```suggestion
DevExtreme DataGrid allows you to create one or multiple AI columns. These columns generate context-based cell values based on component data and a custom prompt, transforming DevExtreme DataGrid into an AI-augmented data exploration tool. In this demo, the <column-name> column (fixed on the right side of the component) is an AI column.
```

Author: A few minor changes, including skipping the column name, it is creatively named "AI Column"))

```suggestion
DevExtreme DataGrid allows you to create one or multiple AI columns. These columns generate meaningful cell values based on component data and a custom prompt, transforming DevExtreme DataGrid into an AI-augmented data exploration tool. In this demo, an AI column is fixed on the right side of the component.
```

Reviewer 2: ok

<!--split-->

[note]

AI services used for this demo have been rate and data limited. As such, you may experience performance-related delays when exploring the capabilities of DataGrid AI Columns.

When connected to your own AI model/service without rate and data limits, DataGrid AI Columns will perform seamlessly, without artificial delays.

[/note]

> This demo implements a default prompt within the AI column. This prompt instructs AI to identify each car's origin country within the DataGrid. You can modify the default prompt or enter a custom prompt in the AI column's header menu.

Reviewer 2: ```suggestion
This demo implements a prompt that instructs AI to identify the origin country of each car. You can modify the default prompt or enter a custom prompt in the AI column header menu.
```

> To integrate an AI column, do the following:

Reviewer 2: ```suggestion
To integrate an AI column into your DevExtreme DataGrid, you must:
```

> - Configure [aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#aiIntegration) (or **columns[]**.**ai**.[aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#aiIntegration)).

Reviewer 2: Check if this is really "option" or "class", or something else.

```suggestion
- Configure [aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#aiIntegration) option at the component level or **columns[]**.**ai**.[aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#aiIntegration) option at the column level.
```

Author: It is a property. I feel adding "at the component/column level" allows us to rephrase this bit a little:
```suggestion
- Configure the **aiIntegration** property at the component or column level ([aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#aiIntegration) or **columns[]**.**ai**.[aiIntegration](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#aiIntegration)).
```

Reviewer 2: good catch!

> - Set a column's [type](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#type) to *"ai"*.

Reviewer 2: ```suggestion
- Set the column [type](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#type) to *"ai"*.
```

Author: At this point, it is "a column" though, users need to set the type property of any column within the component to "ai".

Reviewer 2: ok, let's try. Although if we confugured the column on the previous step, it's "the".

Author: On second thought, my comment is no longer valid now that we added "at the column level" above. I will apply your suggestion, thank you!

> - Specify the column's [name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name).

Reviewer 2: ```suggestion
- Specify the column [name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name).
```

> - Configure **columns[]**.**ai** options, including the generation [mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#mode), predefined [prompt](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#prompt), and [no data text](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#noDataText) (displayed when the AI service returns no data for a row).

Reviewer 2: We've already mentioned the column-level option. What is the difference?

```suggestion
- Configure **columns[]**.**ai** options, including generation [mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#mode), predefined [prompt](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#prompt), and [no data text](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#noDataText) (displayed when the AI service returns no data for a row).
```

Author: Here it is other properties, such as the aforementioned mode, prompt, noDataText, etc.

Author: I think replacing "including" with "such as" may be a better fit here:
```suggestion
- Configure **columns[]**.**ai** options, such as generation [mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#mode), predefined [prompt](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#prompt), and [no data text](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/ai/#noDataText) (displayed when the AI service returns no data for a row).
```

> Our DevExtreme DataGrid component uses all visible row data in AI requests, including unbound and hidden column fields. This data gives LLMs broader context, but increases the component use of AI resources. To limit data included in AI requests, modify the **AIColumnRequestCreatingEvent**.[data](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIColumnRequestCreatingEvent/#data) parameter in the [onAIColumnRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIColumnRequestCreating) event handler.

Reviewer 3: How about just
`uses all row data`
? Just all available data.

> This demo also configures the [onAIColumnRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIColumnRequestCreating) handler to limit data included in AI requests. The default DataGrid behavior is to include all data from visible rows in AI requests, including data not bound to a column and data of hidden columns. This gives LLMs broader context, but increases the component's use of AI resources. To limit data included in AI requests, modify the **AIColumnRequestCreatingEvent**.[data](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIColumnRequestCreatingEvent/#data) parameter.

Reviewer 2: I tried to shorten the paragraph a bit.

```suggestion
Our DevExtreme DataGrid component uses all visible records in AI requests, including unbound and hidden columns. This data gives LLMs broader context, but increases the component use of AI resources. To limit data included in AI requests, modify the **AIColumnRequestCreatingEvent**.[data](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIColumnRequestCreatingEvent/#data) parameter in the [onAIColumnRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIColumnRequestCreating) event handler.
```

Author: Minor data clarification:
```suggestion
Our DevExtreme DataGrid component uses all visible row data in AI requests, including unbound and hidden column fields. This data gives LLMs broader context, but increases the component use of AI resources. To limit data included in AI requests, modify the **AIColumnRequestCreatingEvent**.[data](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIColumnRequestCreatingEvent/#data) parameter in the [onAIColumnRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIColumnRequestCreating) event handler.
```
